src/librustc_typeck/astconv.rs

```diff
@@ -651,11 +651,16 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
         });
         let existential_projections = projection_bounds.iter().map(|bound| {
             bound.map_bound(|b| {
-                let p = b.projection_ty;
+                let trait_ref = self.trait_ref_to_existential(b.projection_ty.trait_ref(tcx));
+                let item_name = tcx.associated_item(b.projection_ty.item_def_id).name;
+                let item_def_id = tcx.associated_items(trait_ref.def_id).find(
+                    |item| item.name == item_name && item.kind == ty::AssociatedKind::Type
+                ).unwrap().def_id;
```

Comment:
In fact, is there no way to convert between a `ProjectionPredicate` and `ExistentialProjection`?
I.e. like there should be between `TraitRef` and `ExistentialTraitRef`. I could be wrong though.